service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceRegistrarProvider.java

```diff
@@ -0,0 +1,25 @@
+package io.smallrye.stork.servicediscovery.kubernetes;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.smallrye.stork.api.ServiceRegistrar;
+import io.smallrye.stork.api.config.ServiceRegistrarAttribute;
+import io.smallrye.stork.api.config.ServiceRegistrarType;
+import io.smallrye.stork.spi.ServiceRegistrarProvider;
+import io.smallrye.stork.spi.StorkInfrastructure;
+
+@ServiceRegistrarAttribute(name = "k8s-host", description = "The Kubernetes API host.")
+@ServiceRegistrarType(value = "kubernetes", metadataKey = KubernetesMetadataKey.class)
+public class KubernetesServiceRegistrarProvider
```

Comment:
Do we need a service registrar for Kube? I would have guessed that the application is automatically registered and nothing need to be done from our side.

Reply:
I have also this remark, shouldn't we register services depending on the service discovery type configured? wdyt @michalszynkiewicz ?

Reply:
I think what Clement says is that for Kube, service registration has limited sense.

Reply:
re having the same configuration for service registration and for service discovery, that's definitely a good idea